msrest/serialization.py

```diff
@@ -1605,12 +1605,15 @@ def deserialize_unicode(data):
     def deserialize_enum(data, enum_obj):
         """Deserialize string into enum object.
 
-        :param str data: response string to be deserialized.
+        If the string is not a valid enum value it will be returned as-is
+        and a warning will be logged.
+
+        :param str data: Response string to be deserialized. If this value is
+         None or invalid it will be returned as-is.
         :param Enum enum_obj: Enum object to deserialize to.
         :rtype: Enum
-        :raises: DeserializationError if string is not valid enum value.
```

Comment:
This is a behavior change (and possibly a breaking change). Maybe we need to revisit the change.

Reply:
Thanks @xiangyan99 - The change to not raise an error here is old - but the docstring was never updated. So _that_ part isn't breaking.

Reply:
It looks to me the old behavior is it returns a valid enum or raise error. And with your change, it may return None?

Reply:
No, the old behaviour to raise an error was removed some time ago, as documented in [this comment here](https://github.com/Azure/msrest-for-python/pull/173/files#diff-3ff87e93ca942963e2ffda1097136bfaR1634).